src/pages/Login/KakaoLogin/KakaoLogin.js

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { CircularProgress } from '@mui/material';
+import axios from 'axios';
+import styled from 'styled-components';
+
+const KakaoLogin = () => {
+  const code = new URL(window.location.href).searchParams.get('code');
+  const navigate = useNavigate();
+  const sendCodeToBack = () => {
+    const option = {
+      url: `http://10.58.5.25:8000/users/kakao/callback?code=${code}`,
+      method: 'GET',
+    };
+    axios(option)
+      .then(res => localStorage.setItem(USER_TOKEN, res.data.token))
+      .then(() => {
+        navigate('/');
+      });
```

Comment:
```suggestion
      .then(res => localStorage.setItem(USER_TOKEN, res.data.token))
      navigate('/')
      );
```

* 통신은 예측이 어렵기때문에, 어떤 response로 들어오는지에 따라 분기처리 해주는게 좋겠습니다.
* .then .then을 2번 할 필요없이 setItem 이후에 바로 navigate 해주면 되겠습니다.(setItem과 navigate 사이에 비동기 처리가 필요해보이진않습니다.)

Reply:
기존 then 2개를 하나로 줄이고 response 분기 처리 후 토큰 저장 -> 페이지 이동으로 수정 하였습니다.